serverless/src/main.rs

```diff
@@ -44,7 +67,61 @@ pub fn main() {
         stats_flusher,
     });
 
-    if let Err(e) = mini_agent.start_mini_agent() {
-        error!("Error when starting serverless trace mini agent: {e}");
+    tokio::spawn(async move {
+        let res = mini_agent.start_mini_agent().await;
+        if let Err(e) = res {
+            error!("Error when starting serverless trace mini agent: {e:?}");
+        }
+    });
+
+    if !dd_use_dogstatsd {
+        return;
```

Comment:
I usually don't love an early return like this, I generally prefer if you'd just wrap the rest in this conditional just so somebody doesn't add more behavior after this line and not realize it isn't executing if statsd is disabled

Reply:
Makes sense, I was trying to reduce the level of nesting but I agree it's more robust to not have the early return.